core/player/src/expressions/evaluator.ts

```diff
@@ -417,4 +457,175 @@ export class ExpressionEvaluator {
       return resolveNode(node.left);
     }
   }
+
+  private async _resolveNodeAsync(
+    _currentValue: any,
+    node: ExpressionNode,
+    options: AsyncHookOptions,
+  ): Promise<unknown> {
+    const { resolveNode, model } = options;
+
+    const expressionContext: ExpressionContext = {
+      ...options,
+      evaluate: (expr) => this.evaluate(expr, options),
+    };
+
+    if (node.type === "BinaryExpression" || node.type === "LogicalExpression") {
```

Comment:
I know the complexity here is having to await the subNode execution but is there any way we could not duplicate the per-node handling logic or abstract that so its reusable somehow?

Reply:
Yeah there's def stuff we can pull out and share -- just have to be careful to not over-complicate it w/ avoiding the async/await